convert.go

```diff
@@ -123,6 +127,16 @@ func DialArgs(m ma.Multiaddr) (string, string, error) {
 				network = "ip4"
 				ip = c.Value()
 				return true
+			case madns.Dns4Protocol.Code:
```

Comment:
Since this PR introduces a dependency to go-multiaddr-dns anyway, wouldn't it be easier to call `madns.Resolve()` on the incoming multiaddr before going into the switch, instead of special casing the DNS protos?

Reply:
Actually, we don't have a context here – that's unfortunate.

Reply:
@raulk see #41 for some discussion around it, but I think we decided that better not. And actually, I don't want it to resolve. I need the hostname back so that a Go http client can then negotiate ssl properly.

Reply:
Got it, thanks for pointing me to the background discussion.